flash/core/model.py

```diff
@@ -583,9 +587,55 @@ def configure_optimizers(self) -> Union[Optimizer, Tuple[List[Optimizer], List[_
             return [optimizer], [self._instantiate_lr_scheduler(optimizer)]
         return optimizer
 
-    @staticmethod
-    def configure_finetune_callback() -> List[Callback]:
-        return []
+    def configure_finetune_callback(
+        self,
+        strategy: Union[str, Tuple[str, int], Tuple[str, Tuple[Tuple[int, int], int]], BaseFinetuning] = "no_freeze",
+        train_bn: bool = True,
+    ) -> List[BaseFinetuning]:
+
+        if isinstance(strategy, BaseFinetuning):
+            return [strategy]
+
+        if isinstance(strategy, str):
+            if strategy not in self.available_finetuning_strategies():
+                raise MisconfigurationException(
+                    f"Please provide a valid strategy from {_DEFAULTS_FINETUNE_STRATEGIES[:3]}."
+                )
+            finetuning_strategy_fn: Callable = self.finetuning_strategies.get(key=strategy)
+            finetuning_strategy_metadata = {"strategy_metadata": None, "train_bn": True}
+
+        elif isinstance(strategy, Tuple):
+            if not isinstance(strategy[0], str) or strategy[0] not in self.available_finetuning_strategies():
+                raise MisconfigurationException(
+                    f"First input of `strategy` in a tuple configuration should be a string within"
+                    f" {_DEFAULTS_FINETUNE_STRATEGIES[3:]}"
+                )
+            if strategy[0] == "freeze_unfreeze" and not isinstance(strategy[1], int):
+                raise MisconfigurationException(
+                    "`freeze_unfreeze` stratgey only accepts one integer denoting the epoch number to switch."
+                )
+            if strategy[0] == "unfreeze_milestones" and not (
+                isinstance(strategy[1], Tuple)
```

Comment:
Could we move the validator to the callback directly?

Reply:
This was initially present in the `trainer.finetune` method just to throw errors before the finetuning starts.
But I moved it to the `Task.configure_finetune_callback` method to maintain the semantics of the code.
Wouldn't pushing to specific callbacks throw the Exceptions a little late in the pipeline ?